tinygrad/ops.py

```diff
@@ -1091,6 +1094,8 @@ def lt_folding(x:UOp, c:int) -> Optional[UOp]:
     return cast(UOp, functools.reduce(operator.add, np).divides(d))<(c//d)
   return None
 
+def two_stage_associative_folding(f: UOp, x: UOp, c1: UOp, c2: UOp): return x.alu(f.op,c1.alu(f.op,c2))
```

Comment:
Why is this not a lambda?

Reply:
Would have to be written in two lines:
```python
  *((UPat.var("x").alu(op, UPat.cvar("c1")).alu(op, UPat.cvar("c2")).named("f"), 
     lambda f,x,c1,c2: x.alu(f.op,c1.alu(f.op,c2))) for op in GroupOp.Associative),
```
Is this preferred? 
Usage of op directly is not possible because it would require closure support (see `deconstruct_function`).